instrumentation/events/lttng-module/net.h

```diff
@@ -148,6 +312,31 @@ LTTNG_TRACEPOINT_EVENT_CLASS(net_dev_template,
 		ctf_integer_hex(void *, skbaddr, skb)
 		ctf_integer(unsigned int, len, skb->len)
 		ctf_string(name, skb->dev->name)
+		ctf_enum(net_network_header, unsigned char, network_header_type, __get_network_header_type(skb))
```

Comment:
That would work only for ipv4 ipv6 protocols, but they are not the only ones (see https://en.wikipedia.org/wiki/Network_layer#Protocols) For others, we would need and enum field anyway. And since we need the enum field, we might as well keep the full ipv4/6 structure together instead of taking out 1 field and having the rest of the header in a variant. But it's as you wish, if you prefer the version field for selector, I can do that.